lsp-fsharp.el

```diff
@@ -57,15 +66,39 @@
 (defun lsp-fsharp--fsac-runtime-cmd ()
   "Get the command required to run fsautocomplete based off of the current runtime."
   (pcase lsp-fsharp-server-runtime
-    ('net-core '("dotnet"))
-    ('mono '("mono"))
+    ('net-core "dotnet")
+    ('mono "mono")
     ('net-framework nil)))
 
+(defun lsp-fsharp--fsac-cmd ()
+  "The location of fsautocomplete executable."
+  (let ((file-ext (if (eq lsp-fsharp-server-runtime 'net-core)
+                      ".dll"
+                    ".exe")))
+    (expand-file-name (concat "fsautocomplete" file-ext) lsp-fsharp-server-install-dir)))
+
+(defun lsp-fsharp--fsac-locate ()
+  "Return the location of the fsautocomplete langauge server."
+  (let ((fsac (lsp-fsharp--fsac-cmd)))
+    (unless (file-exists-p fsac)
+      (if (yes-or-no-p "Server is not installed. Do you want to install it?")
+          (lsp-fsharp--fsac-install)
+        (error "LSP F# cannot be started without FsAutoComplete Server")))
+  fsac))
+
+(defun lsp-fsharp--fsac-install ()
+  "Downloads the latest version of fsautocomplete, and set `lsp-fsharp-server-path'."
+  (let ((temp-file (make-temp-file "fsautocomplete" nil ".zip")))
```

Comment:
Take a look at https://github.com/emacs-lsp/lsp-python-ms/pull/37 which also handles unzipping but it fallbacks to powershell on windows. We probably need to push up that logic in `lsp-mode` and then reuse it in `lsp-python-ms` and `dap-mode`.

Reply:
Good point, fixed.